cdap-support-bundle/bin/observability.py

```diff
@@ -0,0 +1,539 @@
+# Copyright © 2023 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+# observability.py to get logs from CDAP instance deployed on kubernetes.
+
+import argparse
+import base64
+import json
+from json.decoder import JSONDecodeError
+import os
+import shutil
+import subprocess
+import sys
+from datetime import datetime
+
+parser = argparse.ArgumentParser(description='Log collector program for CDAP.')
+parser.add_argument(
+    "--namespace",
+    "-n",
+    help="Kubernetes namespace where CDAP is installed",
+    required=True,
+    type=str
+)
+parser.add_argument(
+    "--instance-name",
+    "-in",
+    help="CDAP instance name",
+    required=True,
+    type=str
+)
+parser.add_argument(
+    "--username",
+    "-u",
+    help="CDAP User Name",
+    type=str
+)
+parser.add_argument(
+    "--password",
+    "-p",
+    help="CDAP Password",
+    type=str
+)
+parser.add_argument(
+    "--cdap-ns",
+    "-cn",
+    help="Cdap namespace",
+    type=str
+)
+parser.add_argument(
+    "--pipeline-name",
+    "-pname",
+    help="CDAP pipeline name",
+    type=str
+)
+parser.add_argument(
+    "--run-id",
+    "-rid",
+    help="CDAP Run id",
+    type=str
+)
+def main(namespace, instance_name, username, password, cdap_ns, pipeline_name, run_id):
+
+    auth_token = ""
+    if username is not None and password is not None:
+        passToken = base64.b64encode(bytes(f'{username}:{password}', 'utf-8'))
+        passToken = passToken.decode('utf-8')
+        output = subprocess.getoutput(f'kubectl -n {namespace} exec -t cdap-{instance_name}-appfabric-0 -c appfabric -- bash -c \'curl -s cdap-{instance_name}-external-authentication:10009/token -H "Authorization:Basic {passToken}"\'')
+        try:
+            str_obj = json.loads(output)
+            auth_token = f"Authorization: Bearer {str_obj['access_token']}"
+        except JSONDecodeError as e:
+            print(e)
+            print('Error : False value of the parameters. Please check if you have provided correct values for all input parameters not able to access the instance.')
```

Comment:
what is "False value of the parameters"? Please fix error message so user can figure out what the issue is, what they need to fix

Reply:
Added the error message. Unable to authenticate due to Invalid credentials. Please check for username/password/namespace in input parameters.